choreolib/src/main/java/choreo/trajectory/SwerveSample.java

```diff
@@ -156,11 +155,40 @@ public SwerveSample interpolate(SwerveSample endValue, double timestamp) {
           MathUtil.interpolate(this.moduleForcesY()[i], endValue.moduleForcesY()[i], scale);
     }
 
+    // Integrate the field speeds to get the pose for this interpolated state, since linearly
+    // interpolating the pose gives an inaccurate result if the speeds are changing between states
+    double lerpedTimestamp = timestamp;
+    double lerpedVXPos = vx;
+    double lerpedVYPos = vy;
+    double lerpedXPos = x;
+    double lerpedYPos = y;
+    double intTime = t + 0.01;
+
+    while (true) {
+      double intT = (intTime - getTimestamp()) / (lerpedTimestamp - getTimestamp());
+      double intAX = MathUtil.interpolate(ax, endValue.ax, intT);
+      double intAY = MathUtil.interpolate(ay, endValue.ax, intT);
+
+      if (intTime >= lerpedTimestamp - 0.01) {
+        double dt = lerpedTimestamp - intTime;
+        lerpedVXPos += intAX * dt;
+        lerpedVYPos += intAY * dt;
+        lerpedXPos += lerpedVXPos * dt;
+        lerpedYPos += lerpedVYPos * dt;
```

Comment:
This using the velocity of the next timestep instead of the current one is suspicious.